apis/beacon/pool/attestations.v2.yaml

```diff
@@ -79,10 +79,7 @@ post:
           anyOf:
             - type: array
               items:
-                $ref: '../../../beacon-node-oapi.yaml#/components/schemas/Attestation'
```

Comment:
so the idea is to always submit SingleAttestation even pre-Electra? in this case should remove the `anyOf`

Reply:
removed

The idea here is that this endpoint should only be used post electra. Since the v2 endpoint is fairly new I thought it'd be ok to repurpose it for post electra usage. Lighthouse is only using the v2 endpoint post electra, but maybe other clients would prefer a v3 endpoint?

Reply:
I don't think we need a v3, this endpoint is effectively only used in devnets right now and would be updated for the next devnet and I don't think anyone is running mixed client setups yet.

In Lodestar we also start using the attestation v2 apis post-electra but what I meant is that after electra is live on mainnet we can be sure that all nodes implement the v2 apis and the v1 apis become unusable since those only support phase0 attestations, this means we could completely remove them from the codebase. We still wanna support earlier forks though, e.g. our sim tests do a fork transitions from phase0 to latest fork, but for that the v2 apis need to be backward compatible.

We should clarify what type of attestation should be submitted pre- and post-electra, submitting `SingleAttestation` should also work for earlier forks, so I don't think this is an issue.

Reply:
Since https://github.com/ethereum/consensus-specs/pull/3900 now also updates the honest validator spec, wouldn't it make the most sense to submit whatever attestation format the validator works with, i.e. `Attestation` for pre-electra and `SingleAttestation` post-electra. This means we don't need to do any transformation before gossiping the attestations.